File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
            long score = 0;
            PriorityQueue<Integer> queue = new PriorityQueue<Integer>(Collections.reverseOrder());
            for(int i=0;i<nums.length;i++){
                queue.add(nums[i]);
            }
            for(int i=0;i<k;i++){
                int temp = queue.poll();
                score += temp;
                queue.add((int)Math.ceil(temp/3.0));
            } 
            return score;
    }
}

File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        nums = [-n for n in nums]
        heapq.heapify(nums)
        score = 0
        for i in range(k):
            temp = -heapq.heappop(nums)
            score += temp
            temp = math.ceil(temp/3)
            heapq.heappush(nums,-temp)
        return score